### ISSUE TITLE
Incorrect error massage when self closing tags are used

### ISSUE THREAD

**Issue by drbcode** [dart-lang/angular_analyzer_plugin#635]
todo_list_component.dart:
```dart
import 'package:angular/angular.dart';
import 'package:angular_components/angular_components.dart';
@Component(
  selector: 'todo-list',
  styleUrls: ['todo_list_component.css'],
  templateUrl: 'todo_list_component.html',
  directives: [],
  providers: [MaterialToggleComponent],
)
class TodoListComponent{}
```

todo_list_component.html:
warning when a self closing tag `<br/>` is used.
```html
<!--warning: Unresolved tag "material-toggle" -->
<br/>
<material-toggle>
</material-toggle>
```

No warning in this case:
```html
<!-- no warning -->
<material-toggle>
</material-toggle>
```

**Comment by zoechi** [comment]
`<br/>` is invalid in HTML5

> There are also tags that are forbidden to be closed: img, input, br, hr, meta, etc.

http://blog.teamtreehouse.com/to-close-or-not-to-close-tags-in-html5

**Comment by drbcode** [comment]
Thanks, then in his case I think error message should be `<br/> is invalid an invalid tag in HTML5` and not `Unresolved tag "material-toggle"` it makes it appear as if there is some problem in `material-toggle` component.

**Comment by drbcode** [comment]
In above case I intentionally passed `MaterialToggleComponent` as a provider and not as a directive.
If `MaterialToggleComponent` is passed as a directive then we don't see any warning for both cases.

**Comment by MichaelRFairhurst** [comment]
@zoechi I don't think you're reading that blog correctly..? (Or I'm not!)

They mention `<br />` specifically:

> An example of a self closing tag is something like a line break (`<br />`)

And when they say "forbidden to be closed" I beleive they are referring to syntax like `<br></br>`., which is not the same as `<br />`, a _self-closing_ tag.

Note that they say:

> the following are both acceptable: `<meta charset="UTF-8">` and `<meta charset="UTF-8" />`

and then say:

> forbidden to be closed: img, input, br, hr, **meta**, etc.

In any case, the error message here would be wrong.

I'll look into this! Thanks for filing!

**Comment by MichaelRFairhurst** [comment]
This title should probably be changed. There _is_ an error in this:

```
import 'package:angular/angular.dart';
import 'package:angular_components/angular_components.dart';
@Component(
  selector: 'todo-list',
  styleUrls: ['todo_list_component.css'],
  templateUrl: 'todo_list_component.html',
  directives: [],
  providers: [MaterialToggleComponent],
)
class TodoListComponent{}
```

should move `MaterialToggleComponent` into `Directives`:

```
import 'package:angular/angular.dart';
import 'package:angular_components/angular_components.dart';
@Component(
  selector: 'todo-list',
  styleUrls: ['todo_list_component.css'],
  templateUrl: 'todo_list_component.html',
  directives: [MaterialToggleComponent],
  providers: [],
)
class TodoListComponent{}
```

Was this an error simply in the filing of this issue, or was this an actual error in the code?

If the latter this would, very strangely, make the issue _backwards_, like a self-closing tag _must_ be used to see an error.

I'm thinking that might be explained by a caching issue or a crash.

**Comment by drbcode** [comment]
In above case I intentionally passed `MaterialToggleComponent` as a provider and not as a directive.
If `MaterialToggleComponent` is passed as a directive then we don't see any warning for both cases.
